pkg/connector/collaborator_cache.go

```diff
@@ -40,9 +44,28 @@ func newCollaboratorCache(workatoClient *client.WorkatoClient, env workato.Envir
 	}
 }
 
+func (p *collaboratorCache) init(ctx context.Context) error {
+	if p.initialized {
+		return nil
+	}
```

Comment:
I think we can remove this first check without the lock

Reply:
@MarcusGoldschmidt updated